lesson-14/src/async.js

```diff
@@ -0,0 +1,26 @@
+// async function fetchPostsById(id) {
+//   const response = await fetch(`https://dummyjson.com/posts/${id}`)
+//   const post = await response.json()
+//   return post
+// }
+
+// async function fetchPosts() {
+//   const posts = await Promise.all([
+//     fetchPostsById(1),
+//     fetchPostsById(2)
+//   ])
+//   return posts
+// }
+
+// async function fetchUsersById(userId) {
+//   const response = await fetch(`https://dummyjson.com/users/${userId}`)
+//   const user = await response.json()
+//   return user
+// }
+
+// fetchPosts()
+//   .then((postsArray) => {
+//     postsArray.map((elementPostArray) => {
+
+//     })
+//   })
```

Comment:
```suggestion
fetchPosts()
  .then(async (postsArray) => {
    console.log(postsArray)

    const userIds = postsArray.map((post) => post.userId)

    console.log('array of user ids', userIds)

    const userFetchPromisesArray = userIds.map((userId) => fetchUsersById(userId))

    console.log('array of promises', userFetchPromisesArray)

    // wait for user fetch promises
    const users = await Promise.all(userFetchPromisesArray)

    console.log(users)

    // map over postsArray and format output
    const postsWithUsers = postsArray.map((post) => ({
      // title
      // user -> find from users variable by post.userId === user.id
    }))

    console.log('posts with users', postsWithUsers)
  })
```

ты почти доделал, осталось

- получить массив из userId
- отправить новые запросы на пользователей
- совместить

добавил чуть направлений